streams/src/main/java/org/apache/kafka/streams/errors/ErrorHandlerContext.java

```diff
@@ -42,6 +42,8 @@ public interface ErrorHandlerContext {
      * {@link org.apache.kafka.streams.kstream.KTable#transformValues(ValueTransformerWithKeySupplier, String...)}
      * (and siblings), that do not always guarantee to provide a valid topic name, as they might be
      * executed "out-of-band" due to some internal optimizations applied by the Kafka Streams DSL.
+     * Additionally, when writing into a changelog topic, there is no associated input record,
+     * and thus no topic name is available.
```

Comment:
Could you be more explicit here?
The error handler context is used in processing exception handler, production exception handler, and deserialization exception handler. It is not immediately clear that this sentence only applies to the production exception handler.

Reply:
> when writing into a changelog topic

Seems pretty clear to me that it's not on the read nor processing path.  -- The statements above area also not more precise. Happy to improve it, but not sure how? Do you have a more concrete suggestions what you want to see (maybe also for the existing statements)?

While I am always in favor of good JavaDocs, also wondering if stuff might out date w/o getting updated when we change things in the runtime. Hence, having some "wiggle room" might actually benefit us?

Reply:
In general I agree, but I would not be too strict about in-line comments -- they can be very helpful to explain important context, which cannot be expressed by code itself IMHO.

I did push another smaller change that I think is valuable for 3.9 release, but will leave it to follow up PRs to improve this further fro 4.0+ release.

Reply:
> they can be very helpful to explain important context, which cannot be expressed by code itself IMHO.

I disagree on this. You can express a lot with code and meaningful names which are better maintanable than inline comments. Of course, there are exceptions.

Reply:
> You can express a lot with code and meaningful names which are better maintanable than inline comments. Of course, there are exceptions.

We are on the same page. I did say, "In general I agree". And I believe, this case in one of the exception you called out :)